src/include/ipc/topology.h

```diff
@@ -34,9 +34,9 @@ enum sof_comp_type {
 	SOF_COMP_MIXER,
 	SOF_COMP_MUX,
 	SOF_COMP_SRC,
-	SOF_COMP_SPLITTER,
+	SOF_COMP_DEPRECATED0, /* Formerly SOF_COMP_SPLITTER */
 	SOF_COMP_TONE,
-	SOF_COMP_SWITCH,
+	SOF_COMP_DEPRECATED1, /* Formerly SOF_COMP_SWITCH */
```

Comment:
these code are totally no harmful, it neither have memory occupation, nor have code size, delete or keep both ok.

Reply:
we cant delete as that changes the IDs, but this change is good for me.

Reply:
Yes, if we deleted it would change the enum and break the ABI, hence deprecation.